stonesoup/initiator/simple.py

```diff
@@ -294,3 +299,214 @@ def initiate(self, detections, timestamp, **kwargs):
+                multi_covar=covar,
+                hypothesis=track.hypothesis)
+
+        return tracks
+
+
+class EnsembleInitiator(GaussianInitiator):
+    """Ensemble State Initiator class
+
+    Utilising Gaussian Initiator, sample from the resultant track's state
+    to generate an Ensemble, overwriting with a
+    :class:`~.EnsembleState`.
+    """
+
+    initiator: GaussianInitiator = Property(
+        doc="Gaussian Initiator which will be used to generate tracks.")
+    ensemble_size: int = Property(
+        default=100,
+        doc="Integer to determine the size of the Gaussian Ensemble State.")
+
+    def __init__(self, *args, **kwargs):
+
+        super().__init__(*args, **kwargs)
+
+        # Create prior particle state
+        try:
+            state = self.initiator.prior_state.state_vector
+            covar = self.initiator.prior_state.covar
+
+        except AttributeError:
+            raise AttributeError("No prior state")
+
+        ensemble = self.create_ensemble(state, covar)
+        self.prior_state = EnsembleState(ensemble)
+
+    def create_ensemble(self, state, covar):
+        """Generates an ensemble based on a provided single Gaussian State
+
+        Parameters
+        ----------
+        state : array of :class:`~.StateVector`
+            The state from which the ensemble will be sampled.
+        covar: array of :class: '~.CovarianceMatrix'
+            The covariance used to generate the ensemble.
+
+        Returns
+        -------
+        : array of :class:`~.StateVectors`
+            The generated ensemble of new state vectors
+        """
+
+        ensemble = []
+        std = np.sqrt(covar)
+        for _ in range(self.ensemble_size):
+            new_state = []
+            for i in range(len(state)):
+                if len(state) == 1:
+                    new_state.append([np.random.normal(state[i], std[i])])
+                else:
+                    new_state.append([np.random.normal(state[i], std[i][i])])
+            ensemble.append(StateVector(new_state))
+        return StateVectors(ensemble)
+
+    def initiate(self, detections, timestamp, **kwargs):
+        """Initiates tracks given unassociated measurements
+
+        Parameters
+        ----------
+        detections : set of :class:`~.Detection`
+            A list of unassociated detections
+        timestamp: datetime.datetime
+            Current timestamp
+
+        Returns
+        -------
+        : set of :class:`~.Track`
+            A list of new tracks with an initial :class:`~.EnsembleState`
+        """
+        tracks = self.initiator.initiate(detections, timestamp, **kwargs)
+
+        for track in tracks:
+            state = track.state_vector
+            covar = track.covar
+            ensemble = self.create_ensemble(state, covar)
+
+            track[-1] = EnsembleStateUpdate(
+                state_vector=ensemble,
+                hypothesis=track.hypothesis,
+                timestamp=track.timestamp)
+
```

Comment:
Would it be possible to use `EnsembleStateUpdate.from_gaussian_state()` and remove the `create_ensemble` method in the initiator? May need to modify the `from_gaussian_state()` to accept kwargs in order to pass the hypothesis.